VMware.CloudFoundation.Reporting.psm1

```diff
@@ -824,7 +824,159 @@ Function Request-SoSHealthJson {
+            $healthSummarySpec.scope.domains[0].domainName = $workloadDomain
+            if ($PSEdition -eq "Core" -and ($PSVersionTable.OS).Split(' ')[0] -eq "Linux") {
+                $reportDestination = ($reportDestination = ($reportPath + "\" + $workloadDomain + "-all-health-results.json")).split('\') -join '/' | Split-Path -NoQualifier
+            } else {
+                $reportDestination = ($reportPath + "\" + $workloadDomain + "-all-health-results.json")
+            }
+        }
+        if (Test-VCFConnection -server $server) {
+            # Create a temporary directory under reportDirectory
+            $createPathCounter = 0
+			for ($createPathCounter -lt 4) {
+				$randomOutput = -join (((48..57)+(65..90)+(97..122)) * 80 |Get-Random -Count 6 |%{[char]$_})
+				$outFilePath = $reportPath + "\" + $randomOutput
+				if (!(Test-Path -Path $outFilePath) -and (Test-Path -Path $reportPath)) {
+					break
+				} else {
+					if ($createPathCounter -eq 3) {
+						Write-Error "Unable to write to $reportPath."
+					}
+					$createPathCounter++
+				}
+			}
+			New-Item -Path $outFilePath -ItemType Directory | Out-NULL
+
+# Use REST API method to request the health summary.			
+			ConvertTo-JSON $healthSummarySpec -depth 10 | Out-File $outFilePath"\healthSummaryPayload.json"
+            $response = Start-VCFHealthSummary -json $outFilePath"\healthSummaryPayload.json"
+			if ($response.id -eq "") {
+				Write-Error "The Health Summary request encountered an issue. Please try again."
+				Return $false
+			}
+			$requestID = $response.id
+			
+            # Retrieve the request status.
+            $response = Get-VCFHealthSummaryTask -id $requestID
+            $escapeCounter = 0
+			While (($escapeCounter -lt 30) -and !($response.status -match "COMPLETED")) {
+				sleep(30)
+				$escapeCounter++
+                $response = Get-VCFHealthSummaryTask -id $requestID
+			}
+			if ($escapeCounter -eq 20) {
+				Write-Error " The Health Summary request is taking an unusual amount of time to complete."
+				Return $false
+			}
+
+            # Download the summary tar.gz to a temporary directory.
+            Request-VCFHealthSummaryBundle -id $requestID
+			$outFile = $outFilePath + "\health-summary.tar.gz"
+			$savedFile = "health-summary-"+$requestID+".tar"
+			if (Test-Path -Path $savedFile) {
+				Copy-Item $savedFile $outFile | Out-NULL
+				Remove-Item -Force $savedFile  | Out-NULL
+			} else {
+				write-error "Encounter issues while downloading HealthSummaryBundle"
+				Return $false
+			}
+
+            # Unzip/untar tar.gz file and extract health-results.json file
+			tar -xzf $outFile -C $outFilePath | Out-NULL
+			$healthSummaryPath = gci -recurse -filter "health-results.json" -Path $outFilePath
+			$healthSummaryFile = $healthSummaryPath.DirectoryName + "\health-results.json"
+			Copy-Item $healthSummaryFile $reportDestination  | Out-NULL
+				
+			# Remove the temporary Directory
```

Comment:
```suggestion
			# Remove the temporary directory.
```

Reply:
addressed in 859f34ae1cb8591be940f19ac2531384052a239b